src/DcmTrajectoryGenerator.cpp

```diff
@@ -0,0 +1,652 @@
+{
+  if (timeBelongsToDomain(t)){
+    // Evaluate the position of the desired DCM at time t
+    double startTime = std::get<0>(m_trajectoryDomain);
+    double time = t - startTime;
+    Eigen::Vector4d tVector;
+    tVector<< 3 * pow(time,2),
+      2 * time,
+      1,
+      0;
+
+    // evaluate booth x and y coordinates
+    double dcmVelX = tVector.dot(m_coefficentsX);
+    double dcmVelY = tVector.dot(m_coefficentsY);
+
+    // store the DCM velocity into the iDyntree vector
+    dcmVel.setVal(0, dcmVelX);
+    dcmVel.setVal(1, dcmVelY);
+    
+    return true;
+  }
+  
+  std::cerr << "[DOUBLE SUPPORT TRAJECTORY] the time t: " << t << "does not belong to the trajectory domain." <<
+    std::endl;
+  return false;  
+}
+
+DcmTrajectoryGenerator::DcmTrajectoryGenerator():
+  m_pauseActive(false)
+{}
+
+DcmTrajectoryGenerator::DcmTrajectoryGenerator(const double &dT, const double &omega):
+  m_omega(omega),
+  m_dT(dT),
+  m_pauseActive(false)
+{}
+
+void DcmTrajectoryGenerator::setOmega(const double &omega)
+{
+  m_omega = omega;
+}
+
+void DcmTrajectoryGenerator::setdT(const double &dT)
+{
+  m_dT = dT;
+}
+
+bool DcmTrajectoryGenerator::addLastStep(const double &singleSupportStartTime,
+					 const double &singleSupportEndTime,
+					 const double &singleSupportDuration,
+					 const double &doubleSupportEndTime,
+					 const double &t0,
+					 const iDynTree::Vector2 &comPosition,
+					 const iDynTree::Vector2 &zmp)
+{
+  // evaluate the Single Support trajectory parameters
+  std::shared_ptr<GeneralSupportTrajectory> newSingleSupport = nullptr;
+  newSingleSupport = std::make_shared<SingleSupportTrajectory>(singleSupportStartTime, singleSupportEndTime,
+							       t0, singleSupportDuration, m_omega,
+							       zmp, comPosition);
+  
+  // instantiate position and velocity boundary conditions vectors
+  iDynTree::Vector2 initPosition, initVelocity;
+  iDynTree::Vector2 endPosition, endVelocity;
+
+  // set boundary conditions
+  if(!newSingleSupport->getDcmPos(singleSupportEndTime, initPosition)){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the position of the DCM in the next SS phase is evaluated." <<std::endl;
+    return false;
+  }
+
+  if(!newSingleSupport->getDcmVel(singleSupportEndTime, initVelocity)){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the velocity of the DCM in the next SS phase is evaluated." <<std::endl;
+    return false;
+  }
+
+  endPosition = comPosition;
+  endVelocity.zero();
+  double doubleSupportStartTime = singleSupportEndTime;
+  std::shared_ptr<GeneralSupportTrajectory> newDoubleSupport = nullptr;
+  newDoubleSupport = std::make_shared<DoubleSupportTrajectory>(initPosition, initVelocity,
+							       endPosition, endVelocity,
+							       doubleSupportStartTime,
+							       doubleSupportEndTime);
+  // add the new Double Support phase
+  m_trajectory.push_back(newDoubleSupport);
+
+  // add the new Single Support phase
+  m_trajectory.push_back(newSingleSupport);
```

Comment:
If it is the **last** step, why are we adding a single support trajectory at the end?

Reply:
Ah no maybe I got it. The vector is actually parsed in reversed order right? (I forgot it in a first place)